server/src/server.js

```diff
@@ -1,337 +1,189 @@
-  let parsed = url.parse(req.url, true),
-    pth = parsed.pathname,
-    query = parsed.query;
-  let backend = "http://" + req.headers.host;
+app.use(function (err, req, res, next) {
+  console.error("Error:", err);
+  console.error(err.stack);
+  errorResponse(res, "General error:", err);
+});
 
-  console.log(req.method, pth, query);
+app.post("/api/register", function (req, res) {
+  let vars = req.params;
+  // FIXME: need to hash secret
+  let canUpdate = vars.userId === req.userId;
+  return registerLogin(vars.userId, {
+    secret: vars.secret,
+    nickname: vars.nickname || null,
+    avatarurl: vars.avatarurl || null
+  }, canUpdate).then(function (ok) {
+    if (ok) {
+      let cookies = new Cookies(req, res, dbschema.getKeygrip());
+      cookies.set("user", vars.userId, {signed: true});
+      simpleResponse(res, "Created", 200);
+    } else {
+      simpleResponse(res, "User exists", 401);
+    }
+  }).catch(function (err) {
+    errorResponse(res, "Error registering:", err);
+  });
+});
 
-  if (req.method === "GET" && (pth.startsWith(jspath) || pth.startsWith(csspath) || pth.startsWith(imgpath) || pth === favicopath)) {
-    let filename = path.join(__dirname, pth);
-    fs.exists(filename, function(exists) {
-      if (exists) {
-        res.setHeader(contentType, lookupContentType(path.extname(filename)));
-        fs.createReadStream(filename).pipe(res);
-      } else {
-        res.writeHead(404); res.end(notFound);
-      }
-    });
-    return;
-  }
-  let modelname = "",
-    storeMap = null;
+app.post("/api/login", function (req, res) {
+  let vars = req.body;
+  checkLogin(vars.userId, vars.secret).then((ok) => {
+    if (ok) {
+      let cookies = new Cookies(req, res, dbschema.getKeygrip());
+      cookies.set("user", vars.userId, {signed: true});
+      simpleResponse(res, "User logged in", 200);
+    } else {
+      simpleResponse(res, "Invalid login", 401);
+    }
+  }).catch(function (err) {
+    errorResponse(err, "Error in login:", err);
+  });
+});
 
-  if (pth.startsWith(registerpath)) {
-    return handleRegister(req, res);
-  } else if (pth.startsWith(loginpath)) {
-    return handleLogin(req, res);
-  }
+app.get("/clip/:id/:domain/:clipId", function (req, res) {
+  let shotId = req.params.id + "/" + req.params.domain;
+  Shot.get(req.backend, shotId).then((shot) => {
+    let clip = shot.getClip(req.params.clipId);
+    if (! clip) {
+      simpleResponse(res, "No such clip", 404);
+      return;
+    }
+    let image = clip.imageBinary();
+    res.setHeader(contentType, image.contentType);
+    res.writeHead(200);
+    res.end(image.data);
+  }).catch((err) => {
+    errorResponse(res, "Failed to get clip", err);
+  });
+});
 
-  if (pth.startsWith(modelspath)) {
-    modelname = pth.slice(modelspath.length);
-    storeMap = models.modelMap;
+app.put("/data/:id/:domain", function (req, res) {
+  let bodyObj = req.body;
+  if (typeof bodyObj == "string") {
+    bodyObj = JSON.parse(bodyObj);
```

Comment:
At some point no?  Or the addon isn't setting Content-Type?  This might no longer be necessary due to changes that happened elsewhere.